dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/DefaultUserService.java

```diff
@@ -687,6 +685,27 @@ private void validateUserRoles(User user, User currentUser, List<ErrorReport> er
     }
   }
 
+  @Override
+  public List<ErrorReport> validateUserRoleCreateOrUpdate(UserRole role, User currentUser) {
```

Comment:
Do we need `  @Transactional(readOnly = true)` for this method ?

Reply:
Possibly, I added it to follow the same pattern as the other validation methods.